src/chrome/internal/breakpoints/bpActionWhenHit.ts

```diff
@@ -0,0 +1,109 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import { IEquivalenceComparable } from '../../utils/equivalence';
+
+/**
+ * These classes represents the different actions that a breakpoint can take when hit
+ * Breakpoint: AlwaysPause
+ * Conditional Breakpoint: ConditionalPause
+ * Logpoint: LogMessage
+ * Hit Count Breakpoint: PauseOnHitCount
+ */
+export interface IBPActionWhenHit extends IEquivalenceComparable {
+    isEquivalentTo(bpActionWhenHit: IBPActionWhenHit): boolean;
+    accept<T>(visitor: IBPActionWhenHitVisitor<T>): T;
+}
+
+export interface IBPActionWhenHitVisitor<T> {
+    alwaysPause(alwaysPause: AlwaysPause): T;
+    conditionalPause(conditionalPause: ConditionalPause): T;
+    pauseOnHitCount(pauseOnHitCount: PauseOnHitCount): T;
+    logMessage(logMessage: LogMessage): T;
+}
+
+abstract class BaseBPActionWhenHit {
+    public abstract accept<T>(visitor: IBPActionWhenHitVisitor<T>): T;
+
+    public isEquivalentTo(bpActionWhenHit: IBPActionWhenHit): boolean {
+        return bpActionWhenHit.accept(new BPActionWhenHitIsEquivalentVisitor(this));
+    }
+}
+
+class BPActionWhenHitIsEquivalentVisitor implements IBPActionWhenHitVisitor<boolean> {
```

Comment:
I don't quite understand what this class is doing

Reply:
This is not necessary any more. Replacing with a simpler version.